File: frontend/src/types/booking.ts
export interface BookingSlot {
  id: number;
  table_id: number;
  arrival_date: string; // ISO datetime
  departure_date: string; // ISO datetime
}

export interface BookingRequest {
  slot_id: number;
  guest_count: number;
}

export interface BookingResponse {
  id: number;
  user_id: string;
  slot_id: number;
  guest_count: number;
  arrival_date: string;
  departure_date: string;
  restaurant_name: string;
}

File: backend/app/schemas/restaurant/booking.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class BookingRequest(BaseModel):
    """Schema for creating a new booking"""

    model_config = ConfigDict(from_attributes=True)

    slot_id: int
    guest_count: int = Field(ge=1, le=100)


class BookingResponse(BaseModel):
    """Schema for booking in API responses"""

    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: str
    slot_id: int
    guest_count: int
    arrival_date: datetime
    departure_date: datetime
    restaurant_name: str
